src/include/duckdb/parser/tableref.hpp

```diff
@@ -37,12 +37,13 @@ class TableRef {
 	shared_ptr<ExternalDependency> external_dependency;
 	//! Aliases for the column names
 	vector<string> column_name_alias;
+	//! User set types for the columns
+	vector<LogicalType> column_type_hint;
```

Comment:
Let's move this to the TableFunctionRef only to minimize serialization challenges